protobuf-lite/src/main/java/io/grpc/protobuf/lite/ProtoLiteUtils.java

```diff
@@ -129,16 +141,21 @@ public T parse(InputStream stream) {
           if (stream instanceof KnownLength) {
             int size = stream.available();
             if (size > 0 && size <= GrpcUtil.DEFAULT_MAX_MESSAGE_SIZE) {
-              byte[] buf = new byte[size];
+              // Coded Input stream does not escape, so buf does not escape.
```

Comment:
Right, which is why the comment is needed here.  We assert that by the end of `parse`, there are no more references to buf (except the thread local)